src/unittest/test.cpp

```diff
@@ -464,6 +510,63 @@ Markdown lol \\|\\|spoiler\\|\\| \\~\\~strikethrough\\~\\~ \\`small \\*code\\* b
 					}
 				});
 			}
+
+			set_test("MEMBER_GET", false);
+			if (!offline) {
+				bot.guild_get_member(TEST_GUILD_ID, TEST_USER_ID, [](const dpp::confirmation_callback_t &event){
+					if (!event.is_error()) {
+						dpp::guild_member m = std::get<dpp::guild_member>(event.value);
+						if (m.guild_id == TEST_GUILD_ID && m.user_id == TEST_USER_ID) {
+							set_test("MEMBER_GET", true);
+						} else {
+							set_test("MEMBER_GET", false);
+						}
+					} else {
+						set_test("MEMBER_GET", false);
+					}
+				});
+			}
+
+			set_test("ROLE_CREATE", false);
+			set_test("ROLE_EDIT", false);
+			set_test("ROLE_DELETE", false);
+			if (!offline) {
+				dpp::role r;
+				r.guild_id = TEST_GUILD_ID;
+				r.name = "Test-Role";
+				r.permissions.add(dpp::p_move_members);
+				r.set_flags(dpp::r_mentionable);
+				r.colour = dpp::colors::moon_yellow;
+				bot.role_create(r, [r, &bot](const dpp::confirmation_callback_t &event) {
+					if (event.is_error()) {
+						return;
+					}
+					dpp::role role = std::get<dpp::role>(event.value);
+					if (role.name == r.name &&
+						role.has_move_members() &&
+						role.flags & dpp::r_mentionable &&
+						role.colour == r.colour) {
+						set_test("ROLE_CREATE", true);
+					}
+					role.guild_id = TEST_GUILD_ID;
+					role.name = "Test-Role-Edited";
+					role.colour = dpp::colors::light_sea_green;
+					bot.role_edit(role, [role_id = role.id, &bot](const dpp::confirmation_callback_t &event) {
+						if (event.is_error()) {
+							return;
+						}
+						dpp::role role = std::get<dpp::role>(event.value);
+						if (role.id == role_id) {
+							set_test("ROLE_EDIT", true);
+						}
+						bot.role_delete(TEST_GUILD_ID, role_id, [](const dpp::confirmation_callback_t &event) {
+							if (!event.is_error()) {
+								set_test("ROLE_DELETE", true);
+							}
```

Comment:
what happens if the role edit fails, so the guild is left created on the test server, the next time the role create runs this will fail permenantly until there is human intervention to delete it... we need to split out these so they arent dependant upon each other, or account for if the role already exists in role create test.

Reply:
True. I didn't think of that. I'll change that tomorrow